core/tests/puppeteer-acceptance-tests/spec/exploration-creator-tests/create-exploration-and-change-basic-settings.spec.ts

```diff
@@ -0,0 +1,116 @@
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Acceptance Test for Exploration Creator and Exploration Manager
+ */
+
+import testConstants from '../../puppeteer-testing-utilities/test-constants';
+import { UserFactory } from '../../puppeteer-testing-utilities/user-factory';
+import { IExplorationCreator } from '../../user-utilities/exploration-creator-utils';
+import { ISuperAdmin } from '../../user-utilities/super-admin-utils';
+
+const DEFAULT_SPEC_TIMEOUT = testConstants.DEFAULT_SPEC_TIMEOUT;
+
+describe('Exploration Creator', function() {
+  let explorationCreator: IExplorationCreator;
+  let explorationVisitor: IExplorationCreator;
+  let superAdmin: ISuperAdmin;
+
+  beforeAll(async function() {
+    explorationCreator = await UserFactory.createNewUser(
+      'explorationAdm', 'exploration_creator@example.com');
+    explorationVisitor = await UserFactory.createNewUser(
+      'explorationVisitor', 'exploration_visitor@example.com');
+    superAdmin = await UserFactory.createNewSuperAdmin('Leader');
+  }, DEFAULT_SPEC_TIMEOUT);
+
+  it('should create an exploration and modify it via the Settings tab',
+    async function() {
+      const guestUser1 = await UserFactory.createNewUser(
+        'guestUsr1', 'guest_user1@example.com',);
+      const guestUser2 = await UserFactory.createNewUser(
+        'guestUsr2', 'guest_user2@example.com');
+      const guestUser3 = await UserFactory.createNewUser(
+        'guestUsr3', 'guest_user3@example.com');
+
+      await guestUser2.closeBrowser();
+      await guestUser1.closeBrowser();
+      await guestUser3.closeBrowser();
+
+      await superAdmin.assignRoleToUser(
+        'explorationAdm', 'voiceover admin');
+      await superAdmin.expectUserToHaveRole(
+        'explorationAdm', 'voiceover admin');
+
+      await superAdmin.assignRoleToUser(
+        'explorationAdm', 'curriculum admin');
+      await superAdmin.expectUserToHaveRole(
+        'explorationAdm', 'curriculum admin');
+
+      await superAdmin.closeBrowser();
+
+      await explorationCreator.openCreatorDashboardPage();
+      await explorationCreator.switchToEditorTab();
+      await explorationCreator.updateCardName('Test question');
+      await explorationCreator.updateExplorationIntroText(
+        'Exploration intro text');
+      await explorationCreator.addEndInteraction();
+      await explorationCreator.showMessageOfSuccessfullExplrationCreation();
+
+      await explorationCreator.goToSettingsTab();
+
+      await explorationCreator.addTitle('Your Title Here');
+      await explorationCreator.expectTitleToHaveMaxLength(36);
+
+      await explorationCreator.updateGoal('OppiaAcceptanceTestsCheck');
+      await explorationCreator.expectGoalToEqual('OppiaAcceptanceTestsCheck');
```

Comment:
@StephenYu2018 there is no need for a goal name or description input, as there is only one input bar for the goal which is quite obvious. Therefore, there is no need to make any changes here. Regarding constants, I believe there is no necessity for them. We can directly use the name of the constant in the string itself. For example, instead of `let goalInput = 'something'`, we can directly write `explorationCreator.updateGoal('goalInput');` and similarly check it in the next line. Please take a look and let's verify with @seanlip.

Reply:
@rahat2134 After some rethinking, I don't think it matters too much whether you use constants or not since `'OppiaAcceptanceTestsCheck'` only used twice as of now. But in general, when there are more instances of the same string literals, it's still best to explain the *intent* of the literal through a variable or constant name. If any dev decides that the data type or value of the literal should change, then  

That being said, I do suggest changing the method name from `updateGoal()` to `updateGoalTo()`, so that anyone reading the top-level tests doesn't run into the same confusion I did, as expressed in my last question at my top paragraph. Top-level acceptance tests are helpful to developers in that the developers understand user flows better after reading the top-level tests. The clarification given was in terms of the exploration editor UI and not every developer who reads the top-level tests may be familiar with that portion of the UI.